spatialos/UnrealGDK
# UNR-2287 Server travel

UNR-2289, UNR-2290 and UNR-2291 are done

UNR-2288 and UNR 2292 are not done

There's some issues with this branch. 
During server travelling clients attempt to use an out of date netdriver (` SetActiveLevelCollection attempted to use an out of date NetDriver: GameNetDriver`)

The GSM never starts accepting players again 


-------

#### Description
Server travel for single server game worlds. 

#### Release note
REQUIRED: Add a release note to the `##Unreleased` section of CHANGELOG.md. You can find guidance for writing useful release notes [here](../SpatialGDK/Extras/internal-documentation/how-to-write-good-release-notes.md). Documentation changes are exempt from this requirement.

#### Tests/QA
1. Open up example project.
2. Add a method of calling server RPC to server travel (something like the nodes below).
![image](https://user-images.githubusercontent.com/56311103/68023929-99fd6c00-fca0-11e9-9067-89d2298d4c1c.png)
3. Click the drop-down on the the play button, click "Advanced Settings". Under "Level Editor/Play/Miltiplayer Options" disable "Use Single Process". Then set the "Editor Multiplayer Mode"
to "Play As Client".
4. Run the client and server by pressing the play button.
5. Trigger the serve travel.

Expected: The client should stay on the currant world for a few moments unable to move. The next map will then be loaded and the lobby screen should be presented. 

#### Documentation
How is this documented (for example: release note, upgrade guide, feature page, in-code documentation)?

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.
